### ISSUE TITLE
fix test failing with MKL

### ISSUE THREAD

**Issue by vigsterkr** [shogun-toolbox/shogun#3992]
when using MKL as BLAS/LAPACK backend then the following tests fails:
```	 42 - unit-LogitVGPiecewiseBoundLikelihood (OTHER_FAULT)
	 93 - unit-CSOBI (OTHER_FAULT)
	201 - unit-PCA (Failed)
	206 - unit-FLDATest (Failed)
	233 - unit-Statistics (Failed)
	278 - integration_meta_cpp-converter-ica_sobi (OTHER_FAULT)
	279 - integration_meta_cpp-converter-ica_fast (Failed)
	310 - integration_meta_cpp-regression-krr_nystrom (Failed)
	368 - libshogun-variational_approx_example (OTHER_FAULT)
	426 - generated_cpp-converter-ica_sobi (OTHER_FAULT)
```

**Comment by vigsterkr** [comment]
@karlnapf fyi: with develop now you can compile shogun with conda's mkl... has some bugs but nothing major...

**Comment by karlnapf** [comment]
This is ace! It deserves a NEWS entry I think. Shall I create a build on buildbot?

**Comment by karlnapf** [comment]
Entrance task:

 * install conda
 * find out how to compile with conda's MKL activated
 * run tests

Most of the failed tests are probably due to slightly changes numerical values when using matrix decompositions.
 
 * integration tests: just don't store the failure causing value in a variable
 * unit tests: investigate

**Comment by vigsterkr** [comment]
`integration_meta_cpp-regression-krr_nystrom` should be fixed

**Comment by newbazz** [comment]
Hey I am new to this organisation, I would like to fix this issue.

**Comment by vigsterkr** [comment]
@newbazz welcome! and welcome to ask if you get stuck :)

**Comment by newbazz** [comment]
@vigsterkr can u help me in solving this issue.

**Comment by vigsterkr** [comment]
@newbazz ? as mentioned in my previous post if you wanna fix this start trying to replicate the problem and then tell me where exactly are you stuck

**Comment by rahul16gupta** [comment]
i want to contribute. please help where to start

**Comment by vigsterkr** [comment]
@rahul16gupta great! as mentioned already above you should compile the latest shogun in develop with MKL support and run the unit tests and you'll get the error.
one way to have intel's MKL is to actually use anaconda.

other than that anybody who's trying to fix this issue plz post here where are you stuck exactly.

**Comment by vigsterkr** [comment]
fixed in #4529
